zcash_client_sqlite/src/lib.rs

```diff
@@ -1425,11 +1483,52 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                                 txout.value,
                                 None,
                             )?;
+
+                            // Even though we know the funding account, we don't know that we have
+                            // information for all of the transparent inputs to the transaction.
+                            #[cfg(feature = "transparent-inputs")]
+                            {
+                                tx_has_wallet_outputs = true;
+                            }
+
+                            // If the decrypted transaction is unmined and has no shielded
+                            // components, add it to the queue for status retrieval.
+                            #[cfg(feature = "transparent-inputs")]
+                            if d_tx.mined_height().is_none() && !detectable_via_scanning {
```

Comment:
Why is this necessary? For an Enhancement request, the tx must have been retrieved via `GetTransaction`, which is supposed to return the current mined height (and despite https://github.com/zcash/lightwalletd/issues/493, we know that it will always return a height > 0 for a mined transaction). So we shouldn't need to make another query.

Reply:
This method makes no assumptions about what has provided it with the transaction data. In particular, this is used in the tests, but it could also be used to introduce a purely-transparent transaction that was added from the mempool.